snapshot/snapshotter.go

```diff
@@ -308,19 +322,14 @@ func checkSnapshotterBasic(t *testing.T, snapshotter Snapshotter, work string) {
 	}
 	defer testutil.Unmount(t, next)
 
-	if err := ioutil.WriteFile(filepath.Join(next, "bar"), []byte("bar\n"), 0777); err != nil {
+	if err := fstest.CheckDirectoryEqualWithApplier(initialApplier, next); err != nil {
 		t.Fatal(err)
```

Comment:
I know this pattern is not used throughout this package but when using the check directory equal function, I find it helpful to fail with a message of what was checked and with `%+v` to pick up as much of the failure trace as possible from `github.com/pkg/errors`